src/react-components/room/RoomSettingsSidebar.js

```diff
@@ -0,0 +1,228 @@
+import React, { useState, useCallback, useEffect } from "react";
+import PropTypes from "prop-types";
+import { useForm } from "react-hook-form";
+import styles from "./RoomSettingsSidebar.scss";
+import { Sidebar } from "../sidebar/Sidebar";
+import { CloseButton } from "../input/CloseButton";
+import { InputField } from "../input/InputField";
+import { IconButton } from "../input/IconButton";
+import { FormattedMessage } from "react-intl";
+import { Button } from "../input/Button";
+import { TextInputField } from "../input/TextInputField";
+import { TextAreaInputField } from "../input/TextAreaInputField";
+import { ToggleInput } from "../input/ToggleInput";
+import { RadioInputField, RadioInputOption } from "../input/RadioInputField";
+import { NumericInputField } from "../input/NumericInputField";
+import { CopyableTextInputField } from "../input/CopyableTextInputField";
+import { BackButton } from "../input/BackButton";
+import { SceneInfo } from "./RoomSidebar";
+
+export function RoomSettingsSidebar({
+  showBackButton,
+  accountId,
+  room,
+  fetchingInvite,
+  inviteUrl,
+  onRevokeInvite,
+  maxRoomSize,
+  showPublicRoomSetting,
+  onSubmit,
+  onClose,
+  canChangeScene,
+  onChangeScene
+}) {
+  const { handleSubmit, register, watch, errors, setValue } = useForm({
+    defaultValues: room
+  });
+
+  const [showRevokeConfirmation, setShowRevokeConfirmation] = useState(false);
+  const revokeInvite = useCallback(() => {
+    setShowRevokeConfirmation(true);
+  }, []);
+  const cancelConfirmRevokeInvite = useCallback(() => {
+    setShowRevokeConfirmation(false);
+  }, []);
+  const confirmRevokeInvite = useCallback(
+    () => {
+      onRevokeInvite();
+      setShowRevokeConfirmation(false);
+    },
+    [onRevokeInvite]
+  );
+
+  const entryMode = watch("entry_mode");
+  const spawnAndMoveMedia = watch("member_permissions.spawn_and_move_media");
+
+  useEffect(
+    () => {
+      if (!spawnAndMoveMedia) {
+        setValue("member_permissions.spawn_camera", false, { shouldDirty: true });
+        setValue("member_permissions.pin_objects", false, { shouldDirty: true });
+      }
+    },
+    [spawnAndMoveMedia, setValue]
+  );
+
+  return (
+    <Sidebar
+      title="Room Settings"
+      beforeTitle={showBackButton ? <BackButton onClick={onClose} /> : <CloseButton onClick={onClose} />}
+    >
+      <form onSubmit={handleSubmit(onSubmit)} className={styles.roomSettingsForm}>
```

Comment:
Have any thoughts on us saving changes immediately instead of requiring a submit button?

Reply:
I think this method is slightly better. Being able to bulk apply changes is probably better and mimics the API a bit better.